test/steps/infinispan.go

```diff
@@ -22,43 +22,64 @@ import (
 )
 
+/*
+	DataTable for Infinispan:
+	| username | developer |
+	| password | mypass    |
+*/
+
 const (
-	//DataTable first column
-	infinispanUsernameKey     = "username"
-	infinispanPasswordNameKey = "password"
+	// DataTable first column
+	infinispanUsernameKey = "username"
+	infinispanPasswordKey = "password"
+
+	externalInfinispanSecret       = "external-infinispan-secret"
+	kogitoExternalInfinispanSecret = "kogito-external-infinispan-secret"
+	usernameSecretKey              = "user"
+	passwordSecretKey              = "pass"
 )
 
+var performanceInfinispanContainerSpec = infinispan.InfinispanContainerSpec{
+	ExtraJvmOpts: "-Xmx2G",
+	Memory:       "3Gi",
+	CPU:          "1",
+}
+
 func registerInfinispanSteps(s *godog.Suite, data *Data) {
 	s.Step(`^Infinispan instance "([^"]*)" is deployed with configuration:$`, data.infinispanInstanceIsDeployedWithConfiguration)
+	s.Step(`^Infinispan instance "([^"]*)" is deployed for performance within (\d+) minute\(s\) with configuration:$`, data.infinispanInstanceIsDeployedForPerformanceWithinMinutesWithConfiguration)
 }
 
 func (data *Data) infinispanInstanceIsDeployedWithConfiguration(name string, table *messages.PickleStepArgument_PickleTable) error {
-	infinispanSecret := "external-infinispan-secret"
+	if err := data.createInfinispanSecret(externalInfinispanSecret, table); err != nil {
+		return err
+	}
+
+	infinispan := framework.GetInfinispanStub(data.Namespace, name, externalInfinispanSecret)
 
-	if err := data.createInfinispanSecret(infinispanSecret, table); err != nil {
+	if err := framework.DeployInfinispanInstance(data.Namespace, infinispan); err != nil {
 		return err
 	}
 
-	infinispan := &infinispan.Infinispan{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: data.Namespace,
-			Name:      name,
-		},
-		Spec: infinispan.InfinispanSpec{
-			Replicas: 1,
-			Security: infinispan.InfinispanSecurity{
-				EndpointSecretName: infinispanSecret,
-			},
-		},
+	return framework.WaitForPodsWithLabel(data.Namespace, "app", "infinispan-pod", 1, 3)
+}
+
+func (data *Data) infinispanInstanceIsDeployedForPerformanceWithinMinutesWithConfiguration(name string, timeOutInMin int, table *messages.PickleStepArgument_PickleTable) error {
+	if err := data.createInfinispanSecret(externalInfinispanSecret, table); err != nil {
+		return err
 	}
+
+	infinispan := framework.GetInfinispanStub(data.Namespace, name, externalInfinispanSecret)
+	// Add performance-specific container spec
+	infinispan.Spec.Container = performanceInfinispanContainerSpec
+
 	if err := framework.DeployInfinispanInstance(data.Namespace, infinispan); err != nil {
 		return err
 	}
 
-	return framework.WaitForPodsWithLabel(data.Namespace, "app", "infinispan-pod", 1, 3)
+	return framework.WaitForInfinispanPodsToBeRunningWithConfig(data.Namespace, performanceInfinispanContainerSpec, 1, timeOutInMin)
 }
 
 func (data *Data) createInfinispanSecret(name string, table *messages.PickleStepArgument_PickleTable) error {
```

Comment:
Do we really need to associate this method to `Data` ?
We are just using the namespace here

Reply:
Good point, that is a leftover from my PR.
Initially I have been considering to have this function as a separate step but then I changed my mind and forgot to remove data. Can be removed (possibly as separate PR).

Reply:
I originally thought the same as Karel, that we want to leave it as a separate step should anybody need it in future :) Resolved now.